device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/twin/TwinGsonBuilder.java

```diff
@@ -0,0 +1,32 @@
+package com.microsoft.azure.sdk.iot.device.twin;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.ToNumberPolicy;
+
+/**
+ * Helper class for building Gson objects to be used when serializing/deserializing twin objects.
+ */
+class TwinGsonBuilder
+{
+    static Gson gson;
+
+    /**
+     * Get the singleton gson instance.
+     * @return the singleton gson instance.
+     */
+    static Gson getGson()
+    {
+        if (gson == null)
```

Comment:
Will thread safety matter, do you think?

Reply:
Not really, no. I can add a comment explaining why